spec/requests/route_spec.rb

```diff
@@ -36,34 +40,39 @@
     end
   end
 
-  describe 'PATCH /user/routes/:id' do
-    let(:route) { create(:route, user: user) }
-
+  describe 'PATCH /user/routes' do
     it 'requires the auth header' do
       patch "/user/routes/#{route.id}", {}, {}
 
       expect(last_response.status).to eq(401)
       expect(parsed_response['errors']).to include('Unauthorized')
     end
 
-    it 'updates the Route' do
-      patch "/user/routes", { id: route.id, mileage: 130, stop: false }.to_json, headers
+    it 'updates the Route with correct mileage' do
+      patch "/user/routes", { id: route.id, mileage: 150, stop: false }.to_json, headers
 
       expect(last_response.status).to eq(200)
-      expect(parsed_response).to include(
-        "data" => data_object_for(route.reload)
-      )
-      expect(route.mileage).to eq(130)
+      expect(parsed_response['data']['route']['mileage']).to eq(150)
     end
 
-    it 'stops the Route' do
-      patch "/user/routes", { id: route.id, mileage: 130, stop: true }.to_json, headers
+    it 'marks the commitment as completed and updates to the next journey' do
```

Comment:
When I read these test descriptions I feel like I am missing the context part. 
```ruby
context 'when the mileage param is present' do 
  it 'updates the route with correct mileage' do
    ...
  end
end

context 'when the route is stopped' do 
  context 'when the route completes the commitment distance' do
  ...
```

Maybe these would read a bit better with that kind of idea in a context or even inline in the description you have here.